src/machine/machine-dbus.c

```diff
@@ -924,9 +925,19 @@ int bus_machine_method_copy(sd_bus_message *message, void *userdata, sd_bus_erro
         if (m->class != MACHINE_CONTAINER)
                 return sd_bus_error_set(error, SD_BUS_ERROR_NOT_SUPPORTED, "Copying files is only supported on container machines.");
 
-        r = sd_bus_message_read(message, "ss", &src, &dest);
-        if (r < 0)
-                return r;
+        if (sd_bus_message_is_method_call(message, NULL, "CopyFromMachineWithFlags") || sd_bus_message_is_method_call(message, NULL, "CopyToMachineWithFlags")) {
+                uint64_t raw_flags;
+                r = sd_bus_message_read(message, "sst", &src, &dest, &raw_flags);
+                if (r < 0)
+                        return r;
+
```

Comment:
You have to check here that there's no undefined flags in the flags read from the message and return an error if there are. If we ignored those we might break users that send undefined flags when we add support for a new flag in the future.